projects/canopy/src/lib/forms/sort-code/sort-code.component.ts

```diff
@@ -120,38 +128,47 @@ export class LgSortCodeComponent implements OnInit, ControlValueAccessor {
       this.ngControl.control.updateValueAndValidity();
     }
 
-    this.valueChanges = this.sortCodeFormGroup.valueChanges.subscribe(
-      (sortCode: SortCodeField) => {
+    this.subscriptions.push(
+      this.sortCodeFormGroup.valueChanges.subscribe((sortCode: SortCodeField) => {
         const first = sortCode.first || '';
         const second = sortCode.second || '';
         const third = sortCode.third || '';
         this.onChange(`${first}${second}${third}`);
-      },
+      }),
+
+      // submit the group when the parent form is submitted
+      this.parentFormGroupDirective.ngSubmit
+        .pipe(filter(({ type }) => type === 'submit'))
+        .subscribe((event) => {
+          this.formGroupDirective.onSubmit(event);
+          this.ngControl.control.updateValueAndValidity();
+        }),
     );
-
-    // submit the group when the parent form is submitted
-    this.parentFormGroupDirective.ngSubmit
-      .pipe(filter(({ type }) => type === 'submit'))
-      .subscribe((event) => {
-        this.formGroupDirective.onSubmit(event);
-        this.ngControl.control.updateValueAndValidity();
-      });
   }
 
   isControlInvalid(control: NgControl, form: FormGroupDirective): boolean {
     return this.errorState.isControlInvalid(control, form);
   }
 
+  ngOnDestroy(): void {
+    this.subscriptions.forEach((subscription) => subscription.unsubscribe());
+  }
+
   writeValue(sortCode: string): void {
     if (!sortCode) {
       return;
     }
 
-    this.sortCodeFormGroup.setValue({
-      first: sortCode.substr(0, 2),
-      second: sortCode.substr(2, 2),
-      third: sortCode.substr(4, 2),
-    });
+    this.sortCodeFormGroup.setValue(
+      {
+        first: sortCode.substr(0, 2),
+        second: sortCode.substr(2, 2),
+        third: sortCode.substr(4, 2),
+      },
+      {
+        emitEvent: false,
```

Comment:
Does this help with the issue you were having with validation? Or is it just to bring it inline with the date field component?

Reply:
No, it doesn't. It's updated mostly because of it being inline but also because it doesn't need to emit an event.